.github/workflows/build.yml

```diff
@@ -404,6 +402,56 @@ jobs:
       - name: Run Checks
         run: ./gradlew test -Dtest.java.version=21
 
+  testopenjdklatest:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master' || contains(github.event.pull_request.labels.*.name, 'jdkversions') || contains(github.event.pull_request.labels.*.name, 'renovate')
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure JDKs
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: |
+            11
+            17
+            23
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Run Checks
+        run: ./gradlew test -Dtest.java.version=23
+
+  testopenjdkearlyaccess:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master' || contains(github.event.pull_request.labels.*.name, 'jdkversions') || contains(github.event.pull_request.labels.*.name, 'renovate')
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
This causes more to be fetched than the default: https://github.com/actions/checkout?tab=readme-ov-file#checkout-v4

Reply:
We use this extensively, I'll cleanup in a separate PR. But leave as is to match other tasks.

Reply:
👍🏻